src/main/kotlin/org/pkl/lsp/VirtualFile.kt

```diff
@@ -197,12 +232,8 @@ class HttpsFile(override val uri: URI, override val project: Project) : BaseFile
     return project.virtualFileManager.get(uri.resolve(path))
   }
 
-  override fun toModule(): PklModule? {
-    return project.fileCacheManager.findHttpModule(uri)
-  }
-
-  override fun contents(): String {
-    return project.fileCacheManager.findHttpContent(uri) ?: ""
+  override fun doReadContents(): String {
+    return uri.toURL().readText()
```

Comment:
We had some checks before in case a https URL was not valid Pkl. I guess we'll just get generic parsing errors now.

Reply:
This will throw `IOException` if the load fails, which we bubble up to `doBuildModule`, which then returns `null`.

It does mean that we will re-fetch HTTP every time, whereas `FileCacheManager` will fail on the first attempt. However, it's possible that a request might fail, but succeed later (no internet connection, etc), so I think retrying is a little better in this case.

Maybe we can throttle how often we retry, to prevent accidentally DOS'ing some poor internet server.

Reply:
Throttle is the best idea but that is an optimization we can do later.

Reply:
Went ahead and added a debouncer (similar to throttle, but new calls push back the timer)